osquery/sql/sqlite_hashing.cpp

```diff
@@ -80,6 +92,141 @@ static void sqliteSsdeepCompareFunc(sqlite3_context* context,
+                                void (*errFunc)(sqlite3_context*,
+                                                const char*)) {
+  // Implemented as defined in https://github.com/corelight/community-id-spec
+
+  const size_t saddr_idx = 0, daddr_idx = 1, sport_idx = 2, dport_idx = 3,
+               proto_idx = 4, seed_idx = 5;
+
+  boost::endian::big_int16_buf_t seed(0);
+  if (argc == 6) {
+    if (sqlite3_value_type(argv[seed_idx]) != SQLITE_INTEGER) {
+      errFunc(context, "Community ID seed must be an integer");
+      return;
+    }
+    const int64_t seed64 =
+        static_cast<int64_t>(sqlite3_value_int64(argv[seed_idx]));
+    if (seed64 < INT16_MIN || seed64 > INT16_MAX) {
+      errFunc(context, "Community ID seed must fit in 2 bytes");
+      return;
+    }
+    seed = seed64;
+  }
+
+  if (sqlite3_value_type(argv[saddr_idx]) != SQLITE_TEXT ||
+      sqlite3_value_type(argv[daddr_idx]) != SQLITE_TEXT) {
+    errFunc(context, "Community ID IPs must be strings");
+    return;
+  }
+  const char* saddr_str =
+      reinterpret_cast<const char*>(sqlite3_value_text(argv[saddr_idx]));
+  const char* daddr_str =
+      reinterpret_cast<const char*>(sqlite3_value_text(argv[daddr_idx]));
+
+  boost::system::error_code ec;
+  ip::address saddr = ip::make_address(saddr_str, ec);
+  if (ec.value() != errc::success) {
+    errFunc(context, "Community ID saddr cannot be parsed as IP");
+    return;
+  }
+  ip::address daddr = ip::make_address(daddr_str, ec);
+  if (ec.value() != errc::success) {
+    errFunc(context, "Community ID daddr cannot be parsed as IP");
+    return;
+  }
+
+  if (sqlite3_value_type(argv[sport_idx]) != SQLITE_INTEGER ||
+      sqlite3_value_type(argv[dport_idx]) != SQLITE_INTEGER) {
+    errFunc(context, "Community ID ports must be integers");
+    return;
+  }
+  const int64_t sport64 =
+      static_cast<int64_t>(sqlite3_value_int64(argv[sport_idx]));
+  const int64_t dport64 =
+      static_cast<int64_t>(sqlite3_value_int64(argv[dport_idx]));
+  if (sport64 < 0 || sport64 > UINT16_MAX || dport64 < 0 ||
+      dport64 > UINT16_MAX) {
+    errFunc(context, "Community ID ports must fit in 2 bytes");
+    return;
+  }
+  boost::endian::big_uint16_buf_t sport(sport64);
+  boost::endian::big_uint16_buf_t dport(dport64);
+
+  if (sqlite3_value_type(argv[proto_idx]) != SQLITE_INTEGER) {
+    errFunc(context, "Community ID protocol must be an integer");
+    return;
+  }
+  const int64_t proto64 =
+      static_cast<int64_t>(sqlite3_value_int64(argv[proto_idx]));
+  if (proto64 < 0 || proto64 > UINT8_MAX) {
+    errFunc(context, "Community ID protocol must fit in 1 byte");
+    return;
+  }
+  uint8_t proto = proto64;
+
+  // Ensure ordering
+  if (!(saddr < daddr || (saddr == daddr && sport64 < dport64))) {
+    std::swap(saddr, daddr);
+    std::swap(sport, dport);
+  }
+
+  // seed . saddr . daddr . proto . 0 . sport . dport
+  std::stringstream bytes;
+  bytes.write(seed.data(), 2);
```

Comment:
linux build:
stderr: osquery/sql/sqlite_hashing.cpp:179:15: error: cannot initialize a parameter of type 'const std::__1::basic_ostream<char, std::__1::char_traits<char> >::char_type *' (aka 'const char *') with an rvalue of type 'unsigned char *'

Reply:
This builds with Buck and CMake in our CI. Maybe you can open a new issue with more details about your environment and/or hop in osquery Slack to debug?